src/util/LazyJsonParser.cpp

```diff
@@ -14,6 +14,21 @@
 
 namespace ad_utility {
 
+// ____________________________________________________________________________
+cppcoro::generator<nlohmann::json> LazyJsonParser::parse(
+    cppcoro::generator<std::string> partJson,
+    std::vector<std::string> arrayPath) {
+  LazyJsonParser p(arrayPath);
+  for (const auto& chunk : partJson) {
+    if (auto res = p.parseChunk(chunk); res.has_value()) {
+      co_yield res;
+      if (p.endReached_) {
```

Comment:
As this is a (albeit `static`) member function of `LazyJsonParser`, `endReached_` can (as all the other member variabls) be private, which then also gets rid of the Sonarcloud message.